### PR TITLE
[MPS] Add logical ops

### PR DESCRIPTION
**Pull request by qqaatw**
This PR adds `logical_not`, `logical_and`, `logical_or`, `logical_xor`.

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/BinaryOps.mm [hunk 1]
```diff
@@ -36,6 +36,12 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha
   const bool is_self_scalar = self.dim() == 0;
   const bool is_other_scalar = other.dim() == 0;
 
+  if (self.numel() > other.numel()){
+    output.resize_(self.sizes());
+  } else {
+    output.resize_(other.sizes());
+  }
```
Comment (qqaatw) [inline aten/src/ATen/native/mps/operations/BinaryOps.mm:43]:
```suggestion
  output.resize_(at::infer_size(self.sizes(), other.sizes()));
```

**Review by albanD — APPROVED** [review]
SGTM
Will let @kulinseth get a second look.

File: aten/src/ATen/native/mps/operations/BinaryOps.mm [hunk 2]
```diff
@@ -36,6 +36,8 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha
   const bool is_self_scalar = self.dim() == 0;
   const bool is_other_scalar = other.dim() == 0;
 
+  output.resize_(at::infer_size(self.sizes(), other.sizes()));
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/operations/BinaryOps.mm:39]: We need to this `resize_` only for the unstructured ops which are being added. We can make this conditional:
```
   auto new_size = at::infer_size(self.sizes(), other.sizes());
   if (!output.is_same_size(new_size)) {
        output.resize_(new_size);
   }
```

**Review by kulinseth — COMMENTED** [review]
Overall it looks good, modulo that comment.

**Review by kulinseth — APPROVED** [review]
Looks good.

**Pull request closed without merging**